apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/first-post-published-modal/index.tsx

```diff
@@ -106,4 +107,12 @@ const FirstPostPublishedModal: React.FC = () => {
 	);
 };
 
+const FirstPostPublishedModal = () => {
+	const { siteIntent: intent } = useSiteIntent();
+	if ( intent === 'write' ) {
```

Comment:
Is there a special reason for us to use this validation outside of the original `FirstPostPublishedModal` function component?

Reply:
I used the same logic used for other modals like `seller-celebration-modal` here: https://github.com/Automattic/wp-calypso/blob/cf98a4b317df30d8f7c5880d628cf2636bee0457/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/seller-celebration-modal/index.jsx#L129

Reply:
As we talked, it seems the advantage of this approach[ is avoiding running all those lines](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/first-post-published-modal/index.tsx#L24C1-L68) before checking the `siteIntent`. I'm good with this strategy.

Out of curiosity @tyxla, can you confirm the strategy we're following here [from the previous SellerCelebrationModal](https://github.com/Automattic/wp-calypso/blob/cf98a4b317df30d8f7c5880d628cf2636bee0457/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/seller-celebration-modal/index.jsx#L129) is a good approach when thinking in performance?

Reply:
No performance concerns from my perspective 👍